lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -412,109 +436,111 @@ withNetworkLayer tr network np project k = do
             let runBF :: forall a. BFM a -> IO a
                 runBF = runBFM bfConfig
             AnyCardanoEra era <- currentNodeEra bfConfig
+            let sp = slottingParameters np
             let stabilityWindow =
-                    fromIntegral . getQuantity $ case cardanoEraStyle era of
-                        LegacyByronEra ->
-                            stabilityWindowByron $ slottingParameters np
-                        ShelleyBasedEra _ ->
-                            stabilityWindowShelley $ slottingParameters np
+                    fromIntegral $ getQuantity case cardanoEraStyle era of
+                        LegacyByronEra -> stabilityWindowByron sp
+                        ShelleyBasedEra _ -> stabilityWindowShelley sp
 
                 isConsensus :: ChainPoint -> IO Bool
-                isConsensus =
-                    runBF . \case
+                isConsensus cp = do
+                    traceWith tr $ MsgDiag "isConsensus"
+                    cp & runBF . \case
                         ChainPointAtGenesis -> pure True
                         ChainPoint (SlotNo slot) blockHeaderHash -> do
                             BF.Block{_blockHash = BF.BlockHash bfHeaderHash} <-
                                 BF.getBlockSlot (BF.Slot (toInteger slot))
                             pure $ bfHeaderHash == toText blockHeaderHash
 
                 getBlockHeaderAtHeight :: Integer -> IO (Maybe BlockHeader)
-                getBlockHeaderAtHeight height =
+                getBlockHeaderAtHeight height = do
+                    traceWith tr $ MsgDiag "getBlockHeaderAtHeight"
                     either (error . show) Just . fromBlockfrost
                         <$> (runBF . BF.getBlock) (Left height)
 
                 getBlockHeaderAt :: ChainPoint -> IO (Maybe BlockHeader)
-                getBlockHeaderAt = runBF . \case
-                    ChainPointAtGenesis -> do
-                        block <-
-                            BF.getBlock $ Right $ genesisBlockHash network
-                        pure . either (error . show) Just $
-                            fromBlockfrost block
-                    ChainPoint (SlotNo slot) blockHeaderHash -> do
-                        b@BF.Block{_blockHash = BF.BlockHash bfHeaderHash} <-
-                            BF.getBlockSlot (BF.Slot (toInteger slot))
-                        pure $
-                            if bfHeaderHash == toText blockHeaderHash
-                                then either (error . show) Just $
-                                        fromBlockfrost b
-                                else Nothing
+                getBlockHeaderAt cp = do
+                    traceWith tr $ MsgDiag "getBlockHeaderAt"
+                    cp & runBF . \case
```

Comment:
```suggestion
                    runBF $ case cp of
```

(… if you like. I just want to point out that once a name is in scope, you can kill two language extensions. 🤓 Similar for other usage sites.)